bundles/SimpleBackendSearchBundle/src/Command/SearchBackendReindexCommand.php

```diff
@@ -46,62 +45,59 @@ protected function execute(InputInterface $input, OutputInterface $output): int
-            $baseClass = Service::getBaseClassNameForElement($type);
-            $listClassName = '\\Pimcore\\Model\\' . $baseClass . '\\Listing';
-            $list = new $listClassName();
-            if (method_exists($list, 'setUnpublished')) {
-                $list->setUnpublished(true);
-            }
-
-            if (method_exists($list, 'setObjectTypes')) {
-                $list->setObjectTypes([
-                    DataObject\AbstractObject::OBJECT_TYPE_OBJECT,
-                    DataObject\AbstractObject::OBJECT_TYPE_FOLDER,
-                    DataObject\AbstractObject::OBJECT_TYPE_VARIANT,
-                ]);
-            }
-
-            $elementsTotal = $list->getTotalCount();
-
-            for ($i = 0; $i < (ceil($elementsTotal / $elementsPerLoop)); $i++) {
-                $list->setLimit($elementsPerLoop);
-                $list->setOffset($i * $elementsPerLoop);
-
-                $this->output->writeln(
-                    'Processing ' .$type . ': ' . ($list->getOffset() + $elementsPerLoop) . '/' . $elementsTotal
-                );
-
-                $elements = $list->load();
-                foreach ($elements as $element) {
-                    try {
-                        //process page count, if not exists
-                        if (
-                            $element instanceof Asset\Document &&
-                            !$element->getCustomSetting('document_page_count') &&
-                            $element->processPageCount()
-                        ) {
-                            $this->saveAsset($element);
-                        }
+            $elementIds = $db->fetchFirstColumn(
+                'WITH RECURSIVE elements AS (
+                    SELECT current.id, current.modificationDate, current.parentId
+                    FROM `'.$type.'s` current
+                    WHERE id=1
+                    UNION ALL
+                    SELECT
+                        descendants.id,
+                        GREATEST(elements.modificationDate, descendants.modificationDate) AS modificationDate,
+                        descendants.parentId
+                    FROM `'.$type.'s` descendants
+                    INNER JOIN elements ON descendants.parentId = elements.id
+                )
+                SELECT elements.id
+                FROM elements
+                LEFT JOIN search_backend_data ON elements.id=search_backend_data.id
+                    AND search_backend_data.mainType=\''.$type.'\'
+                WHERE search_backend_data.id IS NULL
+                    OR search_backend_data.modificationDate < elements.modificationDate'
+            );
+            $elementsTotal = count($elementIds);
+
+            foreach ($elementIds as $i => $elementId) {
+                if ($i % 100 === 0) {
+                    Pimcore::collectGarbage();
+                    Logger::info('Processing '.$type.': '.min($i + 100, count($elementIds)).'/'.$elementsTotal);
```

Comment:
i'd revert `$elementsPerLoop = 100;` and 
```suggestion
                if ($i % $elementsPerLoop === 0) {
                    Pimcore::collectGarbage();
                    Logger::info('Processing '.$type.': '.min($i + $elementsPerLoop, count($elementIds)).'/'.$elementsTotal);
```